cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,441 @@
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Federated": "arn:aws:iam::817312594854:oidc-provider/oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866"
+            },
+            "Action": "sts:AssumeRoleWithWebIdentity",
+            "Condition": {
+                "StringEquals": {
+                    "oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866:sub": "system:serviceaccount:radius-system:applications-rp",
+                    "oidc.eks.us-west-2.amazonaws.com/id/67DDAC18D8C44CEDCF1C9719A8E9B866:aud": "sts.amazonaws.com"
+                }
+            }
+        },
+    ]
+}
+  ```
+
+
+
+the user-specific details would be {ACCOUNT_ID} and {OIDC_PROVIDER}.
+
+
+#### Radius support
+
+**installation and setup**
+
+* use `rad install kubernetes --set global.aws.IRSA.enabled=true` to add the neccessary pod spec to mount the service-account token.
+
+```
+    Containers:
+      ucp:
+        Container ID:   docker://faa5af929f91e053b04b023e05c4fa6363e87688df265443d9f377258fc3fd04
+        Image:          ghcr.io/radius-project/ucpd:latest
+        Image ID:       docker-pullable://ghcr.io/radius-project/ucpd@sha256:991ed703a3260a19535690242cef4d7e4d076038b016443d9106f2d76d57b0da
+        Ports:          9443/TCP, 9090/TCP
+        Host Ports:     0/TCP, 0/TCP
+        :
+        Environment:
+          UCP_CONFIG:                   /etc/config/ucp-config.yaml
+          BASE_PATH:                    /apis/api.ucp.dev/v1alpha3
+          TLS_CERT_DIR:                 /var/tls/cert
+          PORT:                         9443
+        Mounts:
+          /etc/config from config-volume (rw)
+          /var/run/secrets/eks.amazonaws.com/serviceaccount from aws-iam-token (ro)
+          /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-d22ld (ro)
+          /var/tls/cert from cert (ro)
+```
+
+* use `rad credential register aws irsa --iam-role <roleARN>` to register the roleARN that radius would assume for deploying and managing AWS resources.
+This command "PUT"s a Radius Credential Resource. As part of this, the credential gets stored as a kubernetes secret (similar to how Radius manages all credentials today).
+
+* use rad env update to store relevant information for deploying resources to accounts and region associated with specific radius environment. (This step is not newly introduced)
+
+```
+rad env update qa --aws-account-id <aws-account-id> --aws-region <aws-region> 
+```
+
+**post installation**
+
+Once the steps in instalation and setup is completed, UCP should be able to "fetch" the configured credentials and then use AssumeRole to manage AWS resources. AWSCredentialProvider should be updated to support the new credential. This can then be used as part of initializing the UCP AWS Module with IRSA credentials. RP should fetch the credential through UCP and utilize it with Terraform provider.  
```

Comment:
nit

```suggestion
Once the steps in installation and setup is completed, UCP should be able to "fetch" the configured credentials and then use AssumeRole to manage AWS resources. AWSCredentialProvider should be updated to support the new credential. This can then be used as part of initializing the UCP AWS Module with IRSA credentials. RP should fetch the credential through UCP and utilize it with Terraform provider.  
```